app.rb

```diff
@@ -0,0 +1,114 @@
+require_relative 'person'
+require_relative 'student'
+require_relative 'teacher'
+require_relative 'classroom'
+require_relative 'book'
+require_relative 'rental'
+
+class App
+  attr_reader :books, :person, :rentals, :classroom
+
+  def initialize
+    @books = []
+    @person = []
+    @rentals = []
+  end
+
+  def list_all_books
+    puts 'All Books:'
+    @books.each do |book|
+      puts "Title: #{book.title}, Author: #{book.author}"
+    end
+  end
+
+  def list_all_people
+    puts 'All People:'
+    @person.each do |person|
+      puts "#{person.class}, Name: #{person.name}, Age: #{person.age}"
+
+      if person.instance_of?(Teacher)
+        puts "Specialization: [#{person.specialization}]"
+      else
+        puts "Permission: #{person.parent_permission}"
+      end
+    end
+  end
+
+  def create_person
+    puts 'Press 1 for student, 2 for teacher:'
+    is_student = gets.chomp.to_i
+    puts 'Enter person name:'
+    name = gets.chomp
+    puts 'Enter person age:'
+    age = gets.chomp.to_i
+
+    case is_student
+    when 1
+      print 'Does the student have parent permission? [Y/N]: '
+      permission = gets.chomp.downcase == 'y'
+      student = Student.new(1, age, permission, name)
+      @person << student
+    when 2
+      print 'What is the teacher\'s specialization? '
+      specialization = gets.chomp
+      teacher = Teacher.new(age, specialization, name)
+      @person << teacher
+    else
+      puts 'Invalid input'
+      return
+    end
+
+    puts 'Person created successfully.'
+  end
```

Comment:
- [ ] I'm not able to create a person(student or teacher), It's giving the below errors. Please make sure to fix the issue and users should be able to create persons and rentals without any issues.

![image](https://github.com/KAYEMBAJONATHAN/school-library/assets/83803414/93c28968-0c97-44d7-9299-95821290df2c)